### ISSUE TITLE
Bone Dragons don't reduce morale

### ISSUE THREAD

**Issue by Branikolog** [ihhub/fheroes2#985]
In the OG Bone Dragons reduce enemy army's morale.

**Comment by idshibanov** [comment]
@Branikolog This should be already done. They definitely reduce it every time I test. Can you give an example when it's not?

**Comment by Branikolog** [comment]
@idshibanov I simply attack neutral creature with and without Bone Dragons. And both cases these creatures have neutral morale.

**Comment by idshibanov** [comment]
> @idshibanov I simply attack neutral creature with and without Bone Dragons. And both cases these creatures have neutral morale.

Neutral monsters always have neutral morale in OG. It works when you attack bone dragons or you attack a hero.

**Comment by Branikolog** [comment]
@idshibanov, I believe, I used to attack neutral Bone Dragons with my black dragons today. And I still had good morale. Cannot check it now... Let me look again tomorrow to make sure...

**Comment by ihhub** [comment]
Hi @Branikolog , could you please for simplification of developer's work attach a save file to show such behavior? Then we won't have any debates about this :)

**Comment by Branikolog** [comment]
I've checked it again:
Fight against neutral Bone Dragons - Morale of my Unit is still normal.
Some Bone Dragons had joined my army - all neutral creatures still have normal morale.
Fight enemy hero with black dragons - enemy dragons still have normal morale.
[Morale.zip](https://github.com/ihhub/fheroes2/files/4896484/Morale.zip)

**Comment by vincent-grosbois** [comment]
@Branikolog 
I've just checked and the morale penalty is applied (but it triggers like 1 time out of 12 as expected)
What I'm sure sure is :
- tooltip window still show that unit have "normal morale" ie not affected by Bone Dragons
- even dead Bone Dragons will affect morale in battle, I'm not sure it's supposed to work like this or not

**Comment by Branikolog** [comment]
@vincent-grosbois, hi!
Hmm. If negative morale is applied, as you and @idshibanov told before, the problem is just in a tooltip window: every time I check current morale of a creature during battle, it shows normal. While it should be changed to negative state.

**Comment by vincent-grosbois** [comment]
@Branikolog 
ok I think I can change it
What I'm wondering now is if there are other creatures that do this kind of stuff ? (ie changing monster's stats)
Also I wonder if dead dragons count or just alive ones

**Comment by LeHerosInconnu** [comment]
Hello @vincent-grosbois,

> What I'm wondering now is if there are other creatures that do this kind of stuff ? (ie changing monster's stats)

The Bone Dragon is the only creature that changes the stats of opposing (or allied) creatures, but I think the possibility of changing the creature stats by another creature could be considered for Beyond 1.0.

> Also I wonder if dead dragons count or just alive ones

The moral malus of the Bone Dragon's opposing troops is activated at the beginning of the combat and is valid until the end of the combat whether the troop is decimated or not.
The malus counts only once even if there are several troops of Bone Dragons.

However, it can be a game option that the malus only remains applied if there is at least one undecimated troop of Bone Dragons, and the malus would be applied again if a troop of Bone Dragons has been decimated and then reanimated (here again for Beyond 1.0).

**Comment by Branikolog** [comment]
@LeHerosInconnu, thanks for your precision. :) 
I'm really busy these days, so cannot answer in time sometimes. 
@vincent-grosbois, just in case, I want to clarify that Bone Dragons shouldn't affect undead creatures.
